bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/values/MapValueImpl.java

```diff
@@ -210,7 +210,7 @@ public V fillAndGet(Object key) {
 
     @Override
     public Object merge(BMap v2, boolean checkMergeability) {
-        return merge(v2, checkMergeability);
+        return merge((MapValueImpl) v2, checkMergeability);
```

Comment:
Shall we directly use the private Object merge(MapValueImpl v2, boolean checkMergeability) logic here so we can get rid of the casting. we can remove the private method.

Reply:
Yeah, discussed the same with @chiranSachintha offline. The cast to `MapValueImpl` itself seems redundant, since we're casting to `MapValue` later on.

Reply:
This has been done in https://github.com/ballerina-platform/ballerina-lang/pull/26480.